plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestTestingPostgreSqlServer.java

```diff
@@ -61,42 +59,42 @@ public void tearDown()
 
     @Test
     public void testCapturingSuccessfulStatement()
-            throws Throwable
     {
         String sql = "SELECT 1";
-        Set<RemoteDatabaseEvent> remoteDatabaseEvents = captureRemoteEventsDuring(() -> postgreSqlServer.execute(sql));
-        assertThat(remoteDatabaseEvents).contains(new RemoteDatabaseEvent(sql, RUNNING));
+        RemoteDatabaseEvent event = new RemoteDatabaseEvent(sql, RUNNING);
+
+        // verify query was run before
+        assertThat(postgreSqlServer.getRemoteDatabaseEvents()).doesNotContain(event);
+
+        postgreSqlServer.execute(sql);
+        assertEventually(() -> assertThat(postgreSqlServer.getRemoteDatabaseEvents()).contains(event));
```

Comment:
why `assertEventually` here? the query is know to have completed by now.

(if stays, add explanation as code comment)

Reply:
That is because logging events is asynchronous. Query is completed and event is emitted, however it may take some time till the moment we will be able to capture it (goes to log file, so some process has to flush, and I guess reading via docker log API adds a lag too).